src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -341,15 +341,18 @@ class spell_pal_blinding_light : public SpellScript
         return ValidateSpellInfo({ SPELL_PALADIN_BLINDING_LIGHT_EFFECT });
     }
 
-    void HandleDummy(SpellEffIndex /*effIndex*/)
+    void HandleOnHit()
     {
-        if (Unit* target = GetHitUnit())
-            GetCaster()->CastSpell(target, SPELL_PALADIN_BLINDING_LIGHT_EFFECT, true);
+        Unit* caster = GetHitUnit();
+        caster->CastSpell(caster, SPELL_PALADIN_BLINDING_LIGHT_EFFECT, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell()
+        });
     }
 
     void Register() override
     {
-        OnEffectHitTarget += SpellEffectFn(spell_pal_blinding_light::HandleDummy, EFFECT_0, SPELL_EFFECT_APPLY_AURA);
+        OnHit += SpellHitFn(spell_pal_blinding_light::HandleOnHit);
```

Comment:
why not OnEffectHitTarget? I'm curious

Reply:
Yeah, why?

Reply:
I think he might have a mixed though: since the triggered spell uses AoE target as TargetB, he doesn't need to pass Target hook, but the main spell uses TARGET_UNIT_CASTER, so Target hook should be alright here?